ase/io/espresso/_x2y.py

```diff
@@ -105,14 +104,14 @@ def read_pw2wannier_out(fd):
     for line in flines:
         if 'JOB DONE' in line:
             job_done = True
-        if 'PW2WANNIER   :' in line:
+        if line.startswith(calc_class.__name__.upper()):
             time_str = line.split()[-2]
             walltime = time_to_float(time_str)
 
-    calc = PW2Wannier(atoms=structure)
+    calc = calc_class(atoms=structure)
     calc.results['job done'] = job_done
     calc.results['walltime'] = walltime
 
     structure.calc = calc
 
-    yield structure
+    return structure
```

Comment:
@elinscott I had to change it this way because otherwise `pw2wannier` would crash when reading the output. Does not seem that the analogous functions for the other calculators return a generator.. What is the reason for this `yield` here?

Reply:
I find this super-surprising. ASE assumes that these various `read_X_out()` routines return generators because for e.g. molecular dynamics calculations the output files contain several structures (i.e. each time step is a new structure). This means that `read_wann2kcp_out()` and `read_p2wannier_out()` should return generators. I thought that if `read_x2y_out()` finished with `yield structure` and then these two functions simply returned `read_x2y_out(...)` that would do the trick...

Reply:
The double `yield` does not work neither. Also I had tried to put a `return` in `read_x2y_out` and a `yield read_x2y_out` in `read_pw2wannier/wann2kcp_out`. But this has the same effect of the double `return` (so `read_pw2wannier/wann2kcp_out` does not return a generator in that case. That's why I went for the double `return`. What do we do then?

Reply:
To be clear, it's `yield from` in the second instance, not `yield`